components/renku_data_services/notebooks/api/schemas/cloud_storage.py

```diff
@@ -216,6 +216,30 @@ def config_string(self, name: str) -> str:
         """
         if not self.configuration:
             raise ValidationError("Missing configuration for cloud storage")
+
+        # Transform configuration for polybox or switchDrive
+        storage_type = self.configuration.get("type", "")
+        access = self.configuration.get("provider", "")
+
+        if storage_type == "polybox" or storage_type == "switchDrive":
+            self.configuration["type"] = "webdav"
+            self.configuration["provider"] = ""
+
+        if access == "shared" and storage_type == "polybox":
+            self.configuration["url"] = "https://polybox.ethz.ch/public.php/webdav/"
+        elif access == "shared" and storage_type == "switchDrive":
+            self.configuration["url"] = "https://drive.switch.ch/public.php/webdav/"
+        elif access == "personal" and storage_type == "polybox":
+            self.configuration["url"] = "https://polybox.ethz.ch/remote.php/webdav/"
+        elif access == "personal" and storage_type == "switchDrive":
+            self.configuration["url"] = "https://drive.switch.ch/remote.php/webdav/"
+
+        # Extract the user from the public link
+        if access == "shared" and storage_type in {"polybox", "switchDrive"}:
+            public_link = self.configuration.get("public_link", "")
+            user_identifier = public_link.split("/")[-1]
+            self.configuration["user"] = user_identifier
+
```

Comment:
just as a reminder, these changes also need to be backported to https://github.com/SwissDataScienceCenter/renku-notebooks/blob/master/renku_notebooks/api/schemas/cloud_storage.py#L155 to work with Renku v1.